src/controller/python/chip/yaml/YamlParser.py

```diff
@@ -0,0 +1,374 @@
+          'cluster': Name of cluster which to invoke action is targeting.
+          'data_model_lookup': Data model lookup to get attribute object.
+        Raises:
+          ParsingError: Raised if there is a benign error, and there is currently no
+            action to perform for this write attribute.
+          UnexpectedParsingError: Raised if there is an unexpected parsing error.
+        '''
+        super().__init__(item['label'])
+        self._command_name = stringcase.pascalcase(item['command'])
+        self._cluster = cluster
+        self._request_object = None
+        self._expected_raw_response: dict = field(default_factory=dict)
+        self._expected_response_object = None
+
+        command = data_model_lookup.get_command(self._cluster, self._command_name)
+
+        if command is None:
+            raise ParsingError(
+                f'Failed to find cluster:{self._cluster} Command:{self._command_name}')
+
+        command_object = command()
+        if (item.get('arguments')):
+            args = item['arguments']['values']
+
+            request_data_as_dict = YamlUtils.convert_name_value_pair_to_dict(args)
+
+            try:
+                request_data = YamlUtils.convert_yaml_type(
+                    request_data_as_dict, type(command_object))
+            except ValueError:
+                raise ParsingError('Could not covert yaml type')
+
+            # Create a cluster object for the request from the provided YAML data.
+            self._request_object = command_object.FromDict(request_data)
+        else:
+            self._request_object = command_object
+
+        self._expected_raw_response = item.get('response')
+
+        if (self._request_object.response_type is not None and
+                self._expected_raw_response is not None and
+                self._expected_raw_response.get('values')):
+            response_type = stringcase.pascalcase(self._request_object.response_type)
+            expected_command = data_model_lookup.get_command(self._cluster, response_type)
+            expected_response_args = self._expected_raw_response['values']
+            expected_response_data_as_dict = YamlUtils.convert_name_value_pair_to_dict(
+                expected_response_args)
+            expected_response_data = YamlUtils.convert_yaml_type(
+                expected_response_data_as_dict, expected_command)
+            self._expected_response_object = expected_command.FromDict(expected_response_data)
+
+    def run_action(self, dev_ctrl: ChipDeviceCtrl, endpoint: int, node_id: int):
+        try:
+            resp = asyncio.run(dev_ctrl.SendCommand(node_id, endpoint, self._request_object))
+        except chip.interaction_model.InteractionModelError:
+            if (self._expected_raw_response is not None and
+                    self._expected_raw_response.get('error')):
+                logger.debug('Got error response, but was expected')
```

Comment:
I think this might cause issues if someone gives an error: SUCCESS. I know that's the default, but I believe the current yaml parser will accept that.

Reply:
Same with the code below